source/extensions/filters/network/mongo_proxy/config.cc

```diff
@@ -32,7 +33,8 @@ Network::FilterFactoryCb MongoProxyFilterConfigFactory::createFilterFactoryFromP
     fault_config = std::make_shared<Filters::Common::Fault::FaultDelayConfig>(proto_config.delay());
   }
 
-  auto stats = std::make_shared<MongoStats>(context.scope(), stat_prefix);
+  auto stats =
+      std::make_shared<MongoStats>(context.scope(), Stats::Utility::sanitizeStatsName(stat_prefix));
```

Comment:
The fact that extensions have to do this seems kind of fragile to me. Also don't we have this same problem elsewhere like in the HCM? Is there any way we can centralize this in the stats APIs?

Reply:
I thought of just doing the fix inside the symbol table API. 

But it really only comes up in 2 places: Mongo and one other one, where we are both using the macros (needing a trailing dot) and capturing the prefix in a symbol-table, for later join().

I think my preference for these two places would be to eliminate the use of the macros -- which I may cause contentions name-lookups -- at least in those places.

I can look at doing that instead in this PR, or I could do it as a follow-up.

Another cleanup I thought of is to remove the trailing dot from the usage in macros. That would be pretty easy but would hit about 50 places in the code so I'd like that to be a non-functional refactor.

Reply:
> I think my preference for these two places would be to eliminate the use of the macros -- which I may cause contentions name-lookups -- at least in those places.

What do you mean exactly? FWIW the macros are way easier to understand for the average developer IMO than the other options, so I would prefer to keep them where can, as it hides a lot of underlying complexity in the standard case.

> Another cleanup I thought of is to remove the trailing dot from the usage in macros. That would be pretty easy but would hit about 50 places in the code so I'd like that to be a non-functional refactor.

I think this would be my preference. We can just be fully consistent about no dots?

Reply:
I updated to be fully consistent with no trailing dots, and changed the macros to account for that. Actually to avoid blowing up the PR I made it accept either way, with a TODO to remove all trailing dots in a follow-up.